Framework/NexusGeometry/src/NexusGeometrySave.cpp

```diff
@@ -992,14 +1008,54 @@ void saveInstrument(const Geometry::ComponentInfo &compInfo, const Geometry::Det
   // save NXmonitors
   for (size_t index = 0; index < detInfo.size(); ++index) {
     if (detInfo.isMonitor(index)) {
+      // Make spectra detector mappings that can be used
+      std::optional<SpectraMappings> mappings;
+      if (detIdToIndexMap) {
+        mappings = makeMappings(compInfo, *detIdToIndexMap, *indexInfo, *spectrumInfo, detIds, index);
+      }
+
       if (reporter != nullptr)
         reporter->report();
-      writer.monitor(instrument, compInfo, detIds[index], index);
+      writer.monitor(instrument, compInfo, detIds[index], index, mappings);
     }
   }
+}
 
-  file.close(); // close file
+/*
+ * Function: saveInstrument
+ * calls the save methods to write components to file after exception
+ * checking. Produces a Nexus format file containing the Instrument geometry
+ * and metadata.
+ *
+ * @param compInfo : componentInfo object.
+ * @param detInfo : detectorInfo object.
+ * @param fullPath : save destination as full path.
+ * @param rootName : name of root entry
+ * @param logger : logging object
+ * @param append : append mode, means openting and appending to existing file.
+ * If false, creates new file.
+ * @param reporter : (optional) report to progressBase.
+ */
+void saveInstrument(const Geometry::ComponentInfo &compInfo, const Geometry::DetectorInfo &detInfo,
+                    const std::string &fullPath, const std::string &rootName, AbstractLogger &logger, bool append,
+                    Kernel::ProgressBase *reporter) {
+
+  validateInputs(logger, fullPath, compInfo);
+
+  H5::Group parentGroup;
+  H5::H5File file;
+  if (append) {
+    file = H5::H5File(fullPath, H5F_ACC_RDWR); // open file
+    parentGroup = file.openGroup(rootName);
```

Comment:
is it still really 'rootName' here?  Since a lot of that has been migrated to 'parent'

Reply:
You're correct about the name, but I tried to change as few of the _legacy_ names as possible.  Only _some_ of the code was _migrated_ to "parent".
    The original author (I'm speculating) was doing "English as a second language", and used _root_ to mean _parent_ quite a bit.  Unfortunately, they _also_ used "root" to mean "root".

Reply:
**I'll double check and see if I can straighten out (at least) this section of code.**